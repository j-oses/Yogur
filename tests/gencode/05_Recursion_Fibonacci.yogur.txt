 {0}    ssp 9;
 {1}    ujp 34; \\ endFun1
\\ def fibonacci
\\ fun1:
 {2}    ssp 6;
 {3}    sep 3;
 {4}    lda 0 5;
 {5}    ind;
 {6}    ldc 0;
 {7}    equ;
 {8}    lda 0 5;
 {9}    ind;
 {10}   ldc 1;
 {11}   equ;
 {12}   or;
 {13}   fjp 18; \\ else2
 {14}   lda 0 0;
 {15}   ldc 1;
 {16}   sto;
 {17}   ujp 33; \\ endif3
\\ else2:
 {18}   lda 0 0;
 {19}   mst 1;
 {20}   lda 0 5;
 {21}   ind;
 {22}   ldc 1;
 {23}   sub;
 {24}   cup 1 2; \\ fun1
 {25}   mst 1;
 {26}   lda 0 5;
 {27}   ind;
 {28}   ldc 2;
 {29}   sub;
 {30}   cup 1 2; \\ fun1
 {31}   add;
 {32}   sto;
\\ endif3:
 {33}   retf;
\\ endFun1:
 {34}   lda 0 8;
 {35}   ldc 0;
 {36}   sto;
\\ for4:
 {37}   lda 0 8;
 {38}   ind;
 {39}   ldc 7;
 {40}   leq;
 {41}   fjp 59; \\ endFor4
 {42}   lda 0 0;
 {43}   lda 0 8;
 {44}   ind;
 {45}   chk 0 7;
 {46}   ixa 1;
 {47}   mst 0;
 {48}   lda 0 8;
 {49}   ind;
 {50}   cup 1 2; \\ fun1
 {51}   sto;
 {52}   lda 0 8;
 {53}   lda 0 8;
 {54}   ind;
 {55}   ldc 1;
 {56}   add;
 {57}   sto;
 {58}   ujp 37; \\ for4
\\ endFor4:
 {59}   stp;
