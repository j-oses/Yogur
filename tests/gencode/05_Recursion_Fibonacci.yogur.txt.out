Run successfully completed
Final P-machine state:
PC = 59   MP = 0   SP = 8   EP = -1   NP = 100   code size = 60
STORE = 
(0,Int 1)
(1,Int 1)
(2,Int 2)
(3,Int 3)
(4,Int 5)
(5,Int 8)
(6,Int 13)
(7,Int 21)
(8,Int 8)
(9,Bool False)
(10,Int 7)
(11,Int 1)
(12,Int 0)
(13,Int (-1))
(14,Int 51)
(15,Int 7)
(16,Int 10)
(17,Int 21)
(18,Int 8)
(19,Int 0)
(20,Int 10)
(21,Int 24)
(22,Int 31)
(23,Int 5)
(24,Int 18)
(25,Int 8)
(26,Int 3)
(27,Int 0)
(28,Int 18)
(29,Int 32)
(30,Int 31)
(31,Int 3)
(32,Int 26)
(33,Int 3)
(34,Int 1)
(35,Int 0)
(36,Int 26)
(37,Int 40)
(38,Int 31)
(39,Int 1)
(40,Int 34)
(41,Int 1)
(42,Int 1)
(43,Int 33)
(44,Int 47)
(45,Int 31)
(46,Int 0)
(47,Int 41)
(48,Int 1)
(49,Int 1)
(50,Int 53)
(51,Int 31)
(52,Int 0)
(53,Int 47)
(54,Int 1)
(55,Int 1)
(56,Int 59)
(57,Int 31)
(58,Int 0)
(59,Int 53)
(60,Int 1)
(61,Int 1)
(62,Null)
(63,Null)
(64,Null)
(65,Null)
(66,Null)
(67,Null)
(68,Null)
(69,Null)
(70,Null)
(71,Null)
(72,Null)
(73,Null)
(74,Null)
(75,Null)
(76,Null)
(77,Null)
(78,Null)
(79,Null)
(80,Null)
(81,Null)
(82,Null)
(83,Null)
(84,Null)
(85,Null)
(86,Null)
(87,Null)
(88,Null)
(89,Null)
(90,Null)
(91,Null)
(92,Null)
(93,Null)
(94,Null)
(95,Null)
(96,Null)
(97,Null)
(98,Null)
(99,Null)
